### PR TITLE
feat(v2): official CodeSandbox support

### PR DESCRIPTION
**Pull request by sammychinedu2ky**
## Motivation

I saw an issue on adding Codesandbox config for Docusaurus which would enable the creation of Codesandbox templates for Docusaurus.
So I decided to work on that.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

I added a dev script to the package.json file and a sandbox.config.json file for each template under the examples folder
I made use of the classic template to create a docusaurus template on codesandbox.  You can check it out using this link https://codesandbox.io/s/docusaurusclassics-xekwd
I generated the examples folder using this script 
`
mkdir -p examples && cd examples && rm -rf * && npx @docusaurus/init@next init classic classic && npx @docusaurus/init@next init facebook facebook && npx @docusaurus/init@next init bootstrap bootstrap
`
Then added the dev script and sandbox.config.json file manually

## Related PRs

https://github.com/facebook/docusaurus/pull/3702

### CODE REVIEW

**Comment by slorber** [comment]
Your codesandbox seems to work, but for merging this I'll need:
- to have a script in monorepo root package.json that "regenerates" the examples
- not using `npx @docusaurus/init@next init` because this will fetch the init cli script from npm. We'd rather ensure that we use the locally build init script to be sure to run the latest one. If I'm on a branch doing local modifications to a template, I should be able to run the command and see my template modifications appearing immediately in the examples.
- document everything

Please explain with all the details how did you create that codesandbox template. I must be able to test your whole workflow, considering I know nothing about codesandbox.
I tried to import your fork's URL on Codesandbox, but it didn't work and I have no idea what to do about it.

**Comment by sammychinedu2ky** [comment]
Ok cool would work on a script that generates it without npm.
It won't work from. My forks URL because I made use of the myforksurl/examples/classic as the entry point on code sandbox

**Comment by sammychinedu2ky** [comment]
Would detail everything once I make a script

**Comment by slorber** [comment]
Thanks, It  looks like indeed importing an url like https://github.com/sammychinedu2ky/docusaurus/tree/master/examples/facebook seems to work better :)

**Comment by sammychinedu2ky** [comment]
Yes that's is because the entry point has less than 500 modules
But my question is should I create a file specifically for the script .. or should I use the packages docusaurus command to do it locally?

**Comment by slorber** [comment]
Not sure it's the best way, but it seems the following work: `node ./packages/docusaurus-init/bin/index.js init examples/classic classic`

You could add scripts in root package.json that just do that + cleaning folder + copy the required json file

**Comment by sammychinedu2ky** [comment]
Ok thanks would do that🙌🙌

**Comment by sammychinedu2ky** [comment]
I've added a script for that in the package.json file  (examples:generate)

File: examples/classic/package.json
Comment (slorber) [inline examples/classic/package.json:2]: why use 1 space indentation?

File: sandbox.config.json
Comment (slorber) [inline sandbox.config.json:6]: can you generate it in the script instead?

**Review by slorber — CHANGES_REQUESTED** [review]
Thanks, that looks great but can't run `yarn examples:generate` twice without error (need to cleanup)

File: examples/classic/package.json
Comment (sammychinedu2ky) [reply examples/classic/package.json:2]: Sorry about that would change it to 2

File: sandbox.config.json
Comment (sammychinedu2ky) [reply sandbox.config.json:6]: Yh sure let me do that

**Comment by sammychinedu2ky** [comment]
> Thanks, that looks great but can't run `yarn examples:generate` twice without error (need to cleanup)

OK let me work on that

**Comment by sammychinedu2ky** [comment]
I've worked on the fix. Everything is working now.

File: package.json
Comment (slorber) [inline package.json:44]: this should not be in your PR

File: package.json
Comment (slorber) [inline package.json:63]: this should not be in your PR

**Review by slorber — APPROVED** [review]
Thanks for your work, going to merge it.

I did some changes on your code because:
- you reverted some important package.json changes
- your cleanup of example folder did not work reliably
- some ESLint rules didn't pass
- all files should have the FB copyright header
- the node init script should rather forward its logs to parent process, so that users knows something is happening

Otherwise it works fine.

I'd recommend checking your Github workflow, because it's really bad if you end up reverting some changes others have done in your PR,  like it happened here

**Comment by sammychinedu2ky** [comment]
thanks a lot, Sir, I'm grateful for the corrections but I tried pulling from the master branch before creating a new branch locally to work on. I think I made a mistake in that. 
But is there any other thing that made the pull request revert changes others have done?

**Comment by sammychinedu2ky** [comment]
Thanks for merging this pull request too 😁

**Comment by slorber** [comment]
NP

I don't know exactly what happened 😅  just know that some diff was bad and had to manually revert it.

Can you try to document this new Codesandbox support? In Readme and docs etc?

(just wait until I merge https://github.com/facebook/docusaurus/pull/3760)

**Comment by sammychinedu2ky** [comment]
> NP
> 
> I don't know exactly what happened just know that some diff was bad and had to manually revert it.
> 
> Can you try to document this new Codesandbox support? In Readme and docs etc?
> 
> (just wait until I merge #3760)

Ok sure would do that. Thanks again  😁

**Comment by bravo-kernel** [comment]
Awesome addition 👏

**Pull request merged**
